### ISSUE TITLE
Code  details

### ISSUE THREAD

**Issue by YUCOCOCS** [hynnsk/HP#6]
` 
for _im in range(sz×sz):
            if _im == 0:
                index_set = torch.tensor([_im, _im+1, _im+sz, _im+(sz+1)])
            elif _im==(sz-1):
                index_set = torch.tensor([_im-1, _im, _im+(sz-1), _im+sz])
            elif _im==(sz*sz-sz):
                index_set = torch.tensor([_im-sz, _im-(sz-1), _im, _im+1])
            elif _im==(sz*sz-1):
                index_set = torch.tensor([_im-(sz+1), _im-sz, _im-1, _im])
            elif ((1 <= _im) and (_im <= (sz-2))):
                index_set = torch.tensor([_im-1, _im, _im+1, _im+(sz-1), _im+sz, _im+(sz+1)])
            elif (((sz*sz-sz+1) <= _im) and (_im <= (sz*sz-2))):
                index_set = torch.tensor([_im-(sz+1), _im-sz, _im-(sz-1), _im-1, _im, _im+1])
            elif (_im % sz == 0):
                index_set = torch.tensor([_im-sz, _im-(sz-1), _im, _im+1, _im+sz, _im+(sz+1)])
            elif ((_im+1) % sz == 0):
                index_set = torch.tensor([_im-(sz+1), _im-sz, _im-1, _im, _im+(sz-1), _im+sz])
            else:
                index_set = torch.tensor([_im-(sz+1), _im-sz, _im-(sz-1), _im-1, _im, _im+1, _im+(sz-1), _im+sz, _im+(sz+1)])
            self.index_mask[_im][index_set] = 1.
            self.divide_num[_im] = index_set.size(0)

`

Hello, Thank you for your excellent work, I am very interested in your work. I read your code carefully and I don't quite understand some of the above code, can you explain it to me, I look forward to hearing from you!

**Comment by hynnsk** [comment]
Hi, thank you for your interest.

This code is for implementing gradient propagation in Local Hidden Positives (LHP).
Specifically, self.index_mask is a binary mask that is multiplied with each patch feature.
This is done to average the features in a 3x3 region around each patch feature.

Although there can be a better way to implement LHP, we just make a mask manually.
